jsk_recognition_utils/src/pcl/ear_clipping_patched.cpp

```diff
@@ -75,77 +75,56 @@ pcl::EarClippingPatched::triangulate (const Vertices& vertices, PolygonMesh& out
     return;
   }
 
-  std::vector<uint32_t> remaining_vertices (n_vertices);
-  if (area (vertices.vertices) > 0) // clockwise?
-    remaining_vertices = vertices.vertices;
-  else
-    for (int v = 0; v < n_vertices; v++)
+  std::vector<uint32_t> remaining_vertices = vertices.vertices;
+  size_t count = triangulate(remaining_vertices, output);
+
+  // if the input vertices order is anti-clockwise, it always left a
+  // convex polygon and start infinite loops, which means will left more
+  // than 3 points.
+  if(remaining_vertices.size() < 3)return;
```

Comment:
`if (remaining_vertices.size() < 3) return;`

Reply:
Thank you for your review! Updated.